File: graphql-java-annotations/src/main/java/com/example/demo/gql/resolvers/Mutations.java
package com.example.demo.gql.resolvers;

import com.example.demo.gql.types.CreatePostInput;
import com.example.demo.service.PostService;
import graphql.annotations.annotationTypes.GraphQLDescription;
import graphql.annotations.annotationTypes.GraphQLField;
import graphql.annotations.annotationTypes.GraphQLName;
import graphql.annotations.annotationTypes.GraphQLNonNull;
import graphql.kickstart.graphql.annotations.GraphQLMutationResolver;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
@GraphQLMutationResolver
public class Mutations implements ApplicationContextAware {
    static PostService postService;

    @GraphQLField
    @GraphQLNonNull
    @GraphQLDescription("Creates a new post.")
    public static UUID createPost(@GraphQLName("createPostInput") CreatePostInput input) {
        return postService.createPost(input);
    }

    @Override
    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
        postService = applicationContext.getBean(PostService.class);
    }
}

File: graphql-kotlin/src/main/kotlin/com/example/demo/gql/datafetchers/Mutations.kt
package com.example.demo.gql.datafetchers

import com.example.demo.PostService
import com.example.demo.gql.types.Comment
import com.example.demo.gql.types.CommentInput
import com.example.demo.gql.types.CreatePostInput
import com.example.demo.gql.types.Post
import com.expediagroup.graphql.generator.annotations.GraphQLDescription
import com.expediagroup.graphql.server.operations.Mutation
import org.springframework.stereotype.Component
import javax.validation.Valid

@Component
class Mutations(val postService: PostService) : Mutation {

    @GraphQLDescription("Create a new post")
    suspend fun createPost(@Valid input: CreatePostInput): Post {
        return postService.createPost(input)
    }

    @GraphQLDescription("Add comment to post")
    suspend fun addComment(@Valid input: CommentInput): Comment {
        return postService.addComment(input)
    }
}